e2e/test/iothub/properties/PropertiesE2ETests.cs

```diff
@@ -0,0 +1,448 @@
+            serviceTwin.Properties.Reported.Contains(propName1).Should().BeTrue();
+
+            TwinCollection prop1Value = serviceTwin.Properties.Reported[propName1];
+            prop1Value.Contains(propName2).Should().BeTrue();
+
+            string prop2Value = prop1Value[propName2];
+            prop2Value.Should().Be(propValue);
+
+            // Sending a null value for a property will result in service removing the property from the client's twin representation.
+            // For the property patch sent here will result in propName2 being removed.
+            await deviceClient
+                .UpdateClientPropertiesAsync(
+                    new ClientPropertyCollection
+                    {
+                        [propName1] = new Dictionary<string, object>
+                        {
+                            [propName2] = null
+                        }
+                    })
+                .ConfigureAwait(false);
+            serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            serviceTwin.Properties.Reported.Contains(propName1).Should().BeTrue();
+
+            string serializedActualProperty = JsonConvert.SerializeObject(serviceTwin.Properties.Reported[propName1]);
+            serializedActualProperty.Should().Be(propEmptyValue);
+
+            // For the property patch sent here will result in propName1 being removed.
+            await deviceClient
+                .UpdateClientPropertiesAsync(
+                    new ClientPropertyCollection
+                    {
+                        [propName1] = null
+                    })
+                .ConfigureAwait(false);
+            serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            serviceTwin.Properties.Reported.Contains(propName1).Should().BeFalse();
+        }
+
+        private async Task Properties_ClientHandlesRejectionInvalidPropertyNameAsync(Client.TransportType transport)
+        {
+            string propName1 = "$" + Guid.NewGuid().ToString();
+            string propName2 = Guid.NewGuid().ToString();
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            Func<Task> func = async () =>
+            {
+                await deviceClient
+                    .UpdateClientPropertiesAsync(
+                        new ClientPropertyCollection
+                        {
+                            [propName1] = 123,
+                            [propName2] = "abcd"
+                        })
+                    .ConfigureAwait(false);
+            };
+            await func.Should().ThrowAsync<IotHubException>();
```

Comment:
Does this throw a specific code for the exception that we can validate here based on protocol?

Reply:
It does not, no. Since this is mqtt, we don't get the advantage of receiving descriptive error codes from the service.